app/celery/process_ses_receipts_tasks.py

```diff
@@ -205,6 +205,9 @@ def process_ses_results(  # noqa: C901 (too complex 14 > 10)
             )
             return
 
+        if incoming_status in (NOTIFICATION_PERMANENT_FAILURE, NOTIFICATION_DELIVERED):
+            notification.personalisation = {k: '<redacted>' for k in notification.personalisation}
```

Comment:
This works, but it's less performant than iteration because it creates a new dictionary in memory and orphans the existing value, which will need to be garbage collected.
```suggestion
            for key in notification.personalisation:
                notification.personalisation[key] = '<redacted>'
```

Reply:
This fails the test for some reason.  It probably has something to do with how sessions work.  Let's just move on.